src/main/java/net/snowflake/ingest/SimpleIngestManager.java

```diff
@@ -61,6 +66,12 @@
     // the key pair we want to use to authenticate
     private KeyPair keypair;
 
+    // Used to add additional information to User-Agent(Http Header)
+    private String userAgentSuffix;
+
+    // Hostname to connect to, default will be RequestBuilder#DEFAULT_HOST
+    private String hostName;
+
```

Comment:
When do we use this inner Builder class and why these two fields are necessary here?

Reply:
Builder patterns makes it easier to construct an object. Having multi argument constructor can easily fade into confusions regarding what each parameter accepts. https://www.vogella.com/tutorials/DesignPatternBuilder/article.html#:~:text=The%20builder%20pattern%20simplifies%20the,to%20create%20an%20immutable%20class.